waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -29,11 +28,15 @@ import qualified Wasp.Util.IO as IOUtil
 analyzeWaspProject ::
   Path' Abs (Dir WaspProjectDir) ->
   CompileOptions ->
-  IO (Either [CompileError] AS.AppSpec)
-analyzeWaspProject waspDir options = runExceptT $ do
-  waspFilePath <- ExceptT $ Control.Arrow.left pure <$> findWaspFile waspDir
-  declarations <- ExceptT $ analyzeWaspFileContent waspFilePath
-  ExceptT $ constructAppSpec waspDir options declarations
+  IO (Either [CompileError] AS.AppSpec, [CompileWarning])
+analyzeWaspProject waspDir options = do
+  findWaspFile waspDir >>= \case
+    Left e -> return (Left [e], [])
+    Right waspFilePath -> do
+      analyzeWaspFileContent waspFilePath >>= \case
+        Left es -> return (Left es, [])
+        Right declarations ->
+          constructAppSpec waspDir options declarations
```

Comment:
I needed more flexibility because return type changed, now I am also piping warnings through, so I found it really hard to write that with ExceptT. If you know how to write it nicely with ExceptT, I am happy to look at it, but I couldn't figure it out and concluded that the solution is probably too complex.